diff --git a/bin/formatjs b/bin/formatjs
index 3d968a46111f1a130bde589a7d9eb64d67799686..fa82dfa08c3f42f5c316dd8a840a50c1a39b249f 100755
--- a/bin/formatjs
+++ b/bin/formatjs
@@ -163516,8 +163516,11 @@ function calculateLineColFromOffset(text, start) {
 function isBuiltinFormatter(format) {
 	return !format || typeof format === "string" && BUILTIN_FORMATTERS.has(format);
 }
+// https://github.com/formatjs/formatjs/issues/6730
+function canNativeExtractIdInterpolationPattern(pattern) {
+	return /^\[sha512:contenthash:(?:base64|base64url|base62|hex):[1-9]\d*\]$/i.test(pattern);
+}
 function canExtractWithNative(files, opts, readFromStdin) {
-	return !readFromStdin && typeof opts.idInterpolationPattern === "string" && !opts.ast && !opts.extractSourceLocation && !opts.onMetaExtracted && !opts.onMsgError && !opts.onMsgExtracted && !opts.overrideIdFn && !opts.pragma && !opts.removeDefaultMessage && isBuiltinFormatter(opts.format) && files.every((file) => existsSync(file) && NATIVE_SUPPORTED_EXTENSIONS.has(extname(file).toLowerCase()));
+	return !readFromStdin && typeof opts.idInterpolationPattern === "string" && canNativeExtractIdInterpolationPattern(opts.idInterpolationPattern) && !opts.ast && !opts.extractSourceLocation && !opts.onMetaExtracted && !opts.onMsgError && !opts.onMsgExtracted && !opts.overrideIdFn && !opts.pragma && !opts.removeDefaultMessage && isBuiltinFormatter(opts.format) && files.every((file) => existsSync(file) && NATIVE_SUPPORTED_EXTENSIONS.has(extname(file).toLowerCase()));
 }
 async function processFile(source, fn, { idInterpolationPattern, ...opts }) {
 	let messages = [];
