diff --git a/dist/index.esm.mjs b/dist/index.esm.mjs
index 929aa56640e2afb00626455e3f61bc79c38c9f97..7bae6a00ce61ac9be7aa9c644b77ef2155096a7b 100644
--- a/dist/index.esm.mjs
+++ b/dist/index.esm.mjs
@@ -294,6 +294,8 @@ var generateWatchOutput = (names, _names, formValues, isGlobal, defaultValue) =>
 };
 
 var isPrimitive = (value) => isNullOrUndefined(value) || !isObjectType(value);
+var isTypedArray = (value) =>
+    ArrayBuffer.isView(value) && !(value instanceof DataView);
 
 function deepEqual(object1, object2, _internal_visited = new WeakSet()) {
     if (isPrimitive(object1) || isPrimitive(object2)) {
@@ -302,6 +304,14 @@ function deepEqual(object1, object2, _internal_visited = new WeakSet()) {
     if (isDateObject(object1) && isDateObject(object2)) {
         return object1.getTime() === object2.getTime();
     }
+    if (isTypedArray(object1) || isTypedArray(object2)) {
+        if (!(isTypedArray(object1) && isTypedArray(object2))) return false;
+        if (object1.length !== object2.length) return false;
+        for (var i = 0; i < object1.length; i++) {
+            if (object1[i] !== object2[i]) return false;
+        }
+        return true;
+    }
     const keys1 = Object.keys(object1);
     const keys2 = Object.keys(object2);
     if (keys1.length !== keys2.length) {
