diff --git a/dist/index.esm.mjs b/dist/index.esm.mjs
index 20f3313bc779a7ebe165ebd612bdc9bc8987ca47..f23c019a25b99c901d36873ff86c1e111b142b62 100644
--- a/dist/index.esm.mjs
+++ b/dist/index.esm.mjs
@@ -674,6 +674,8 @@ var createSubject = () => {
 };
 
 var isPrimitive = (value) => isNullOrUndefined(value) || !isObjectType(value);
+var isTypedArray = (value) =>
+    ArrayBuffer.isView(value) && !(value instanceof DataView);
 
 function deepEqual(object1, object2, _internal_visited = new WeakSet()) {
     if (isPrimitive(object1) || isPrimitive(object2)) {
@@ -682,6 +684,14 @@ function deepEqual(object1, object2, _internal_visited = new WeakSet()) {
     if (isDateObject(object1) && isDateObject(object2)) {
         return object1.getTime() === object2.getTime();
     }
+    if (isTypedArray(object1) || isTypedArray(object2)) {
+        if (!(isTypedArray(object1) && isTypedArray(object2))) return false;
+        if (object1.length !== object2.length) return false;
+        for (var i = 0; i < object1.length; i++) {
+            if (object1[i] !== object2[i]) return false;
+        }
+        return true;
+    }
     const keys1 = Object.keys(object1);
     const keys2 = Object.keys(object2);
     if (keys1.length !== keys2.length) {
